File: tools/wave/src/utils/serializer.js
const Session = require("../data/session");

/**
 * @module Serializer
 */
class Serializer {
  /**
   * @param {Session[]} sessions
   */
  serializeSessions(sessions) {
    return sessions.map(session => this.serializeSession(session));
  }

  /**
   * @param {Session} session
   */
  serializeSession(session) {
    const token = session.getToken();
    const tests = session.getTests();
    const types = session.getTypes();
    const user_agent = session.getUserAgent();
    const timeouts = session.getTimeouts();
    const test_files_count = session.getTestFilesCount();
    const test_files_completed = session.getTestFilesCompleted();
    const pending_tests = session.getPendingTests();
    const running_tests = session.getRunningTests();
    const completed_tests = session.getCompletedTests();
    const status = session.getStatus();
    const date_started = session.getDateStarted();
    const date_finished = session.getDateFinished();
    const browser = session.getBrowser();
    const is_public = session.isPublic();
    const reference_tokens = session.getReferenceTokens();
    const webhook_urls = session.getWebhookUrls();
    const expiration_date = session.getExpirationDate();
    return {
      token,
      tests,
      types,
      user_agent,
      timeouts,
      test_files_count,
      test_files_completed,
      pending_tests,
      running_tests,
      completed_tests,
      status,
      browser,
      date_started,
      date_finished,
      is_public,
      reference_tokens,
      webhook_urls,
      expiration_date
    };
  }
}

let serializer = new Serializer();
module.exports = serializer;

File: tools/wave/utils/serializer.py
from __future__ import unicode_literals
from datetime import datetime


def serialize_session(session):
    return {
        "token": session.token,
        "types": session.test_types,
        "user_agent": session.user_agent,
        "labels": session.labels,
        "timeouts": session.timeouts,
        "test_state": session.test_state,
        "last_completed_test": session.last_completed_test,
        "tests": session.tests,
        "pending_tests": session.pending_tests,
        "running_tests": session.running_tests,
        "status": session.status,
        "browser": session.browser,
        "date_created": session.date_created,
        "date_started": session.date_started,
        "date_finished": session.date_finished,
        "is_public": session.is_public,
        "reference_tokens": session.reference_tokens,
        "webhook_urls": session.webhook_urls,
        "expiration_date": session.expiration_date,
        "type": session.type,
        "malfunctioning_tests": session.malfunctioning_tests
    }

def serialize_sessions(sessions):
    serialized_sessions = []
    for session in sessions:
        serialized_sessions.append(serialize_session(session))
    return serialized_sessions

def serialize_device(device):
    return {
        "token": device.token,
        "user_agent": device.user_agent,
        "name": device.name,
        "last_active": datetime.utcfromtimestamp(device.last_active/1000).isoformat()
    }
